### PR TITLE
Simplify `create_instance` calls

### LINKED ISSUE #5523: Simplify create_instance calls
Calling https://github.com/scrapy/scrapy/blob/bbfa185664cef79299b48cb0ae22065439bb07fc/scrapy/utils/misc.py#L144 with a crawler is needlessly verbose:

```python
create_instance(objclass, settings=None, crawler=crawler)
create_instance(objclass, settings=crawler.settings, crawler=crawler)  # functionally identical
```

It would be great if we could support a shorter alternative:

```python
create_instance(objclass, crawler=crawler)
```

It is not trivial to do in a backward-compatible way, given the function signature.

It may be also a good idea to log deprecation warnings when `settings` and `crawler` are used as positional arguments, as well as when `objcls` is used as a named parameter, so that we can eventually stop supporting that and move to a better signature:

```python
def create_instance(objcls, /, *args, crawler=None, settings=None, **kwargs): 
```

Although, looking at it now, I realize that such a signature would cause `settings` and `crawler` to be after `*args`, which could be confusing. Maybe it would be best to go for separate functions:

```python
def build_from_crawler(objcls, crawler, /, *args, **kwargs):
def build_from_settings(objcls, settings, /, *args, **kwargs):
```

### PR DESCRIPTION
**Pull request by jxlil**
Added `build_from_crawler` and `build_from_settings` functions

Fixes #5523, closes #5884, closes #6169.

### CODE REVIEW

File: scrapy/utils/misc.py [hunk 1]
```diff
@@ -160,6 +160,50 @@ def create_instance(objcls, settings, crawler, *args, **kwargs):
...
+def build_from_crawler(objcls, crawler, /, *args, **kwargs):
+    """Construct a class instance using its ``from_crawler`` constructor, if
+    available.
+
+    The arguments ``*args`` and ``**kwargs`` are forwarded to the constructors.
+
+    Raises ``TypeError`` if the resulting instance is ``None`` (e.g. if an
+    extension has not been implemented correctly.)
+    """
+    if crawler and hasattr(objcls, "from_crawler"):
```
Comment (Gallaecio) [inline scrapy/utils/misc.py:172]:
```suggestion
    if hasattr(objcls, "from_crawler"):
```

I think that only made sense when settings could be specified and crawler not.

File: scrapy/utils/misc.py [hunk 2]
```diff
@@ -160,6 +160,50 @@ def create_instance(objcls, settings, crawler, *args, **kwargs):
...
+        instance = objcls.from_crawler(crawler, *args, **kwargs)
+        method_name = "from_crawler"
+    else:
```
Comment (Gallaecio) [inline scrapy/utils/misc.py:175]: In this function, if `from_settings` exists, it should be used with `crawler.settings`.

**Review by Gallaecio — COMMENTED** [review]
It is a good first step, but there is some more work to do, such as:
- Logging a deprecation warning when `create_instance` is called.
- Replacing all usages of `create_instance` in the code base by one of these new functions.

**Comment by jxlil** [comment]
Hi @Gallaecio, thanks for responding, I'm going to close this PR.

**Pull request closed without merging**